server.R

```diff
@@ -18,6 +18,103 @@
 # ---------------------------------------------------------
 
 server <- function(input, output, session) {
+  # Loading screen ---------------------------------------------------------------------------
+  # Call initial loading screen
+
+  hide(id = "loading-content", anim = TRUE, animType = "fade")
+  show("app-content")
+
+
+  # cookie consent ----------------------------------------------------------
+
+  observeEvent(input$cookies, {
+    if (!is.null(input$cookies)) {
+      if (!("dfe_analytics" %in% names(input$cookies))) {
+        shinyjs::show(id = "cookieMain")
+      } else {
+        msg <- list(
+          name = "dfe_analytics",
+          value = input$cookies$dfe_analytics
+        )
+        session$sendCustomMessage("analytics-consent", msg)
+        if ("cookies" %in% names(input)) {
+          if ("dfe_analytics" %in% names(input$cookies)) {
+            if (input$cookies$dfe_analytics == "denied") {
+              ga_msg <- list(name = paste0("_ga_", google_analytics_key))
```

Comment:
Looks like the google_analytics_key hasn't been set. Needs to be defined in global.R and should be the digits from the GA tag excluding the "G-" from the start.

Reply:
Just pushed https://github.com/dfe-analytical-services/Transition-matrices-dashboard/pull/12/commits/90779c5ffc439ed8c5fcafd265d7a5e7fb8c8e26  - is that what you meant?

Reply:
Yep, looks good